### ISSUE TITLE
crash on init when asked for backend value

### ISSUE THREAD

**Issue by blalor** [hashicorp/terraform#20667]
When running `terraform init` and hitting `^C` when asked for a configuration value for a backend, or running `terraform init -input=false` when using a partially-configured backend, Terraform crashes.

### Terraform Version
```
Terraform v0.12.0-beta1
```

### Terraform Configuration Files
```hcl
terraform {
  backend "atlas" {
    # address = …
    # access_token = …
    # name = …
  }
}
```

Full config archive: [crasher.zip](https://github.com/hashicorp/terraform/files/2960931/crasher.zip)

### Debug Output

1. [with interrupt](https://github.com/hashicorp/terraform/files/2960954/trace.log)
2. [with `-input=false`](https://github.com/hashicorp/terraform/files/2960958/trace.log)

### Crash Output

1. [with interrupt](https://github.com/hashicorp/terraform/files/2960966/crash.log)
2. [with `-input=false`](https://github.com/hashicorp/terraform/files/2960962/crash.log)

### Expected Behavior

I should have gotten an error like that from Terraform v0.11.13:

```
Initializing the backend...
access_token
  Access token to use to access Atlas. If ATLAS_TOKEN is set then
  this will override any saved value for this.

  Enter a value: ^C
Error asking for input to configure the backend "atlas": access_token: interrupted
```

### Actual Behavior

Crash.

```
Initializing the backend...
name
  Full name of the environment in Terraform Enterprise, such as 'myorg/myenv'

  Enter a value: ^C
panic: value is unknown

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.AsString(...)
	/opt/gopath/pkg/mod/github.com/zclconf/go-cty@v0.0.0-20190201220620-4ca19710f056/cty/value_ops.go:1009
github.com/hashicorp/terraform/backend/atlas.(*Backend).PrepareConfig(0xc00054d7a0, 0x29f9c00, 0xc00054a2b8, 0x2406c20, 0x387dda0, 0x1, 0xc0002aefb0, 0x1, 0x1, 0x387dda0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/atlas/backend.go:81 +0x82a
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc0002fd380, 0xc0000b2820, 0xc00041a03e, 0xc000538c48, 0x1033673, 0x0, 0xc000538c58, 0x1032dac, 0xc000000300, 0x300000002, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:1043 +0x2bf
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_s(0xc0002fd380, 0xc0000b2820, 0xb4ea4d9e, 0xc000165500, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:613 +0x6f
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc0002fd380, 0xc0002cabc0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:462 +0xd68
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc0002fd380, 0xc0002cabc0, 0x0, 0xc000539300, 0x1, 0x1, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:85 +0x5d1
github.com/hashicorp/terraform/command.(*InitCommand).initBackend(0xc0002fd380, 0xc000102cb0, 0x26c32d2, 0xf, 0xc0002c5f80, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/command/init.go:450 +0x1ab
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc0002fd380, 0xc00000a090, 0x0, 0x0, 0xc0002ae130)
	/opt/gopath/src/github.com/hashicorp/terraform/command/init.go:287 +0x18ac
github.com/mitchellh/cli.(*CLI).Run(0xc0004b0280, 0xc0004b0280, 0xc000417d20, 0x1)
	/opt/gopath/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x207
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:223 +0xb37
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:100 +0xb7
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:36 +0x2e



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

#### interrupt

1. run `terraform init`
2. hit `^C` when asked for a value

#### without input

1. run `terraform init -input=false`

**Comment by jbardin** [comment]
Thanks @blalor,

This was fixed in #20546

**Comment by blalor** [comment]
Oops! Sorry for the dupe!

—
Brian Lalor
blalor@hashicorp.com
